proposals/Exceptions.md

```diff
@@ -170,21 +174,104 @@ also be sure to also pop off the caught exception values.
 
 ### Rethrows
 
-The _rethrow_ instruction has no arguments, and can only appear in catch
-blocks. It always re-throws the exception caught by the catch block. This allows
-the catch block to clean up state before the exception is passed back to the
-next enclosing try block.
+The _rethrow_ instruction can only appear in the body of a catch block.  The
+_rethrow_ instruction always re-throws the exception caught by the referenced
+catch block. This allows the catch block to clean up state before the exception
+is passed back to the next enclosing try block.
+
+Associated with the _rethrow_ instruction is a _label_. The label is used to
+disambiguate which exception is to be rethrown, when inside nested catch blocks.
+
+The label is the relative block depth to the corresponding try block for which
+the catching block appears.
+
+For example consider the following example:
+
+```
+try
+  ...
+catch 1
+  ...
+  begin
+    ...
+    try
+      ...
+    catch 2
+      ...
+      try
+        ...
+      catch 3
+        ...
+        rethrow N
+      end
+    end
+  end
+  ...
+end
+```
+
+In this example, `N` is used to disambiguate which caught exception is being
+rethrown. It could rethrow any of the three caught expceptions. Hence,
+`rethrow 0` corresponds to the exception caught by `catch 3`, `rethrow 1`
+corresponds to the exception caught by `catch 2`, and `rethrow 4` corresponds
+to the exception caught by `catch 1`.
 
```

Comment:
Added comment describing why `rethrow 2` is not allowed because it doesn't reference a try block. Also fixed `rethow 4` which should be `rethrow 3`.